app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -1,32 +1,68 @@
 package com.lightricks.feedexercise.ui.feed
 
+import android.annotation.SuppressLint
+import android.content.Context
+import android.util.Log
 import androidx.lifecycle.*
 import com.lightricks.feedexercise.data.FeedItem
+import com.lightricks.feedexercise.data.FeedRepository
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.network.FeedApiService
 import com.lightricks.feedexercise.util.Event
-import java.lang.IllegalArgumentException
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
 
 /**
  * This view model manages the data for [FeedFragment].
  */
-open class FeedViewModel : ViewModel() {
+open class FeedViewModel(context: Context) : ViewModel() {
+    private var disposableRequest: Disposable? = null
+    private val repository: FeedRepository =
+        FeedRepository(FeedDatabase.build(context), FeedApiService())
```

Comment:
`FeedRepository` is supposed to have a different lifecycle than `FeedViewModel`. If we will add more `Fragments` to the app we don't want to recreate the repository for each screen. 

In other words we want to inject `FeedRepository` into  `FeedViewModel`. And the simplest way to do it is probably using `FeedViewModelFactory`.

Reply:
Yep, done on 1.1. Nothing like manual lifecycling for pure-model. Saves a lot of trouble.